linux/Platform.c

```diff
@@ -192,6 +192,22 @@ double Platform_setCPUValues(Meter* this, int cpu) {
    }
    percent = CLAMP(percent, 0.0, 100.0);
    if (isnan(percent)) percent = 0.0;
+
+   v[CPU_METER_FREQUENCY] = 0;
+   if (this->pl->settings->showCPUFrequency) {
+      char filename[63+1];
+      xSnprintf(filename, 63, "/sys/devices/system/cpu/cpu%d/cpufreq/scaling_cur_freq", cpu - 1);
```

Comment:
@Arnavion 
Are you sure that `cpu - 1` is still correct when choosing to show CPU numbers up from `0`? I could not find it quickly now, if this `cpu` input value is changed right before printing the numbers to the meter or this value is changed in the very first place, so that it can be `0` already and would be reduced to `-1` then, invalid and all others associated with the wrong core then.

In case a quick test with CPU numbers up from `0` should be easy.

Reply:
Yes, I'm sure. I tested it already. The `cpu == 0` case is always for the Average CPU meter.